### ISSUE TITLE
twisted.protocols.test.test_basic doesn't following coding standard

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6042]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6042|
|Type|defect|
|Created|2012-10-03 12:27:06Z|
|Branch|https://github.com/twisted/twisted/tree/coding-standard-6042|

`twisted.protocols.test.test_basic` (new in [#6025](https://github.com/twisted/twisted/issues/6025), but just moved existing code mostly) has a variety of issues: bad test names, missing docstrings, uses fail variations of assertion methods, etc.. It should be updated.

Attachments:

* [my-twisted-patch-6042.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ea8/ea84b497dc48869bc490a090094cb9e417e28765/ed49af79f99dda0d7b1882dac5f8342a31e7790b.patch) (8303 bytes) - added by esacteksab on 2013-02-09 08:25:19Z - 
* [my-twisted-patch-6042-3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ea8/ea84b497dc48869bc490a090094cb9e417e28765/013e6bbf3fb911f4a1dd9dd51e762bfe2f5b2284.patch) (8347 bytes) - added by esacteksab on 2013-02-10 02:38:16Z - 
* [my-twisted-patch-6042-4.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ea8/ea84b497dc48869bc490a090094cb9e417e28765/37e1007b807788a4a9157aef8dabcf3861accb9b.patch) (8359 bytes) - added by esacteksab on 2013-02-13 01:35:25Z - This one passes tests
* [6042_introom.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ea8/ea84b497dc48869bc490a090094cb9e417e28765/7c37dc639a34b9bb0ab951942a694c51f4a2ce0f.patch) (6377 bytes) - added by introom on 2013-04-16 13:58:50Z - 
* [6042_introom_2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ea8/ea84b497dc48869bc490a090094cb9e417e28765/978a009c2cbd07d2ccfb895c40bc4ed7ffc9714e.patch) (5517 bytes) - added by introom on 2013-04-18 15:10:15Z - 
* [6042_introom_3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ea8/ea84b497dc48869bc490a090094cb9e417e28765/7f76685d02b373f9e283e6584009792cf5724b9e.patch) (5211 bytes) - added by introom on 2013-04-18 15:15:18Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6042 6042
type__defect defect
reporter__itamar itamar
priority__low low
milestone__ 
branch__branches_coding_standard_6042 branches/coding-standard-6042
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1349267226000000 1349267226000000
changetime__1397528601000000 1397528601000000
version__None None
owner__tom_prince tom.prince
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="esacteksab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|esacteksab commented|
|-|-|

Only my second ever patch. I'm certain it's not complete nor perfect. Seeking guidance/direction. 

Removed failIf, chopped off at 80 chars, added some doc strings. 

I have doubts about rcvd for received (not sure if rcvd is OK, or if RX would work or ?)

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince set owner to @tomprince|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to esacteksab|
|-|-|

Thanks for your contribution. Here are a few things I saw at a first read through:

  1. Test method names should start with `test_` but otherwise follow the naming standard. That is, start with a lowercase letter and be camel case, e.g. `test_pauseResume` not `test_PauseResume`.
  1. Normally, test docstrings should describe the expected behavior they are testing.
  1. `recvd` is actually a (deprecated) attribute for some things in `twisted.protocol.basic`.
  1. There are some logic changes in `test_PauseResume` which also lacks a docstring.
  1. `assertTrue(...)` is preferable to `assertEqual(..., True)`.

**Comment by twisted-trac** [comment]
|<img alt="esacteksab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|esacteksab commented|
|-|-|

1. changed ```test_PauseResume``` to ```test_pauseResume```

2. hopefully I got this docstring right. 

3. I know there is a depreciation policy for backwards compatibility. Does this test still need to exist?

4. Added docstring to ```test_pauseResume```

5. Replaced ```assertTrue(...)``` with ```assertEqual(..., True)```.

**Comment by twisted-trac** [comment]
|<img alt="esacteksab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|esacteksab removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="aal95's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|aal95 set owner to aal95<br>aal95 set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_8"></a>|@jerub commented|
|-|-|

Sorry, things have changed and this patch no longer applies cleanly. Can you refresh and resubmit the patch please?

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|introom removed owner<br>introom set status to `new`|
|-|-|

added some docstring
modified the code to be within 80 columns
deleted test_deprecatedModuleAttributes per ticket[#6321](https://github.com/twisted/twisted/issues/6321)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

(In [38136]) Branching to 'coding-standard-6042'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra set owner to introom|
|-|-|

Replying to [introom](#note_9):
> added some docstring
> modified the code to be within 80 columns
> deleted test_deprecatedModuleAttributes per ticket[[#6321](https://github.com/twisted/twisted/issues/6321)](https://github.com/twisted/twisted/issues/6321)

Thanks for your patch. I'm not sure why the test is being deleted here, and not in #6321? Your patch is also missing some of the improvements made by esacteksab in [[my-twisted-patch-6042-4.patch],](None/my-twisted-patch-6042-4.patch],) could you include those as well and supply a new patch?

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_12"></a>|introom commented|
|-|-|

Replying to [thijs](#note_11):
> Replying to [introom](#note_9):
> > added some docstring
> > modified the code to be within 80 columns
> > deleted test_deprecatedModuleAttributes per ticket[[#6321](https://github.com/twisted/twisted/issues/6321)](https://github.com/twisted/twisted/issues/6321)
> 
> Thanks for your patch. I'm not sure why the test is being deleted here, and not in #6321? Your patch is also missing some of the improvements made by esacteksab in [[my-twisted-patch-6042-4.patch],](None/my-twisted-patch-6042-4.patch],) could you include those as well and supply a new patch?

Thanks for your review. 
But I am not sure which improvement I am missing. 
Say in line 1052 (http://twistedmatrix.com/trac/attachment/ticket/6042/my-twisted-patch-6042-4.patch) I think it's not suitable to add p.pauseProducing. because if the code is ok, then p.pauseProducing should already been invoked in p.dataReceived(str) in line 1050.

Would you please tell me which improvement I am missing?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra commented|
|-|-|

Replying to [introom](#note_12):
> 
> Would you please tell me which improvement I am missing?

I meant the changes he did in test_basic relating to getting things on lines of 80, and adding docstrings to test(cases) that don't have them (for example `ConsumingProtocol` and `ProducerTestCase`).

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_14"></a>|introom commented|
|-|-|

Replying to [thijs](#note_13):
> Replying to [introom](#note_12):
> > 
> > Would you please tell me which improvement I am missing?
> 
> I meant the changes he did in test_basic relating to getting things on lines of 80, and adding docstrings to test(cases) that don't have them (for example `ConsumingProtocol` and `ProducerTestCase`).
Hi
As to 80 cols, in  http://twistedmatrix.com/trac/attachment/ticket/6042/my-twisted-patch-6042-4.patch
i.e. the first change in line 57, actually it's exactly 79cols < 80cols, so I think's unnecessary to make it span two lines.

As to docstrings, I re-compared my patch and with 6042-4, but find they are basically the same.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra commented|
|-|-|

Replying to [introom](#note_14):
> As to docstrings, I re-compared my patch and with 6042-4, but find they are basically the same.

Ok, could you refresh the patch against trunk now that [#6321](https://github.com/twisted/twisted/issues/6321) landed? And the docstrings should use `"""` instead of `'''`.

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_16"></a>|introom removed owner|
|-|-|

added an empty .misc

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_17"></a>|@thijstriemstra commented|
|-|-|

(In [38157]) apply 6042_introom_2.patch, refs [#6042](https://github.com/twisted/twisted/issues/6042)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_18"></a>|@tomprince set owner to @tomprince<br>@tomprince set status to `assigned`|
|-|-|

`testPauseResume`s name doesn't follow the coding standard, and its docstring doesn't explain what the test is trying to assert.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_19"></a>|@tomprince commented|
|-|-|

(In [38504]) Add some explantion to testPauseResume.

Refs: [#6042](https://github.com/twisted/twisted/issues/6042).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_20"></a>|@tomprince removed owner<br>@tomprince set status to `new`|
|-|-|

I renamed the test method, and add some more detailed explanation of what `testPauseResume` is trying to test.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_21"></a>|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

reviewing...

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_22"></a>|@wallrj set owner to @tomprince<br>@wallrj set status to `new`|
|-|-|

## Code Review

Notes:
 * Merges cleanly
 * Build results look ok:
   * http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/coding-standard-6042
 * Formatting and whitespace changes make things more readable.
 * New test comments make the intent much clearer.
 * Coverage is slightly improved
   ```
   [richard@zorin Twisted]$ diff -u trunk/.coverage.report branches/coding-standard-6042/.coverage.report
   --- trunk/.coverage.report	2013-06-19 08:45:50.079155125 +0100
   +++ branches/coding-standard-6042/.coverage.report	2013-06-19 08:46:37.706934850 +0100
   @@ -1,3 +1,3 @@
    Name                      Stmts   Miss  Cover
    ---------------------------------------------
   -twisted/protocols/basic     301     16    95%
   +twisted/protocols/basic     308     16    95%
   ```


Some things to fix or answer:
 1. twisted/protocols/test/test_basic.py
    1. Typo: "writes that that line"
    2. There are still a few docstrings that say "Test x" and "x
       should y"
    3. Various Twistedchecker errors, some of which look valid:
       ```
       ************* Module twisted.protocols.test.test_basic
       W9208:1050,4:OnlyProducerTransport.pauseProducing: Missing docstring
       W9208:1054,4:OnlyProducerTransport.resumeProducing: Missing docstring
       W9208:1058,4:OnlyProducerTransport.write: Missing docstring
       W9402:1187,0: The first letter of comment should be capitalized
       W9013:152,0: Expected 3 blank lines, found 2
       C0103:184,12:LineReceiverTestCase.test_buffer: Invalid name "packet_size" (should match ([a-z_][a-zA-Z0-9]*)$)
       W9015:194,0: Too many blank lines, expected 2
       C0103:205,12:LineReceiverTestCase.test_pausing: Invalid name "packet_size" (should match ([a-z_][a-zA-Z0-9]*)$)
       C0103:228,12:LineReceiverTestCase.test_rawPausing: Invalid name "packet_size" (should match ([a-z_][a-zA-Z0-9]*)$)
       C0103:249,12:LineReceiverTestCase.test_stopProducing: Invalid name "packet_size" (should match ([a-z_][a-zA-Z0-9]*)$)
       W9208:405,0:TestMixin: Missing docstring
       W9208:407,4:TestMixin.connectionMade: Missing docstring
       W9208:411,4:TestMixin.stringReceived: Missing docstring
       W9208:418,4:TestMixin.connectionLost: Missing docstring
       W9208:423,0:TestNetstring: Missing docstring
       W9208:431,0:LPTestCaseMixin: Missing docstring
       W9204:438,4:LPTestCaseMixin.getProtocol: Missing epytext markup @return for return value
       C0103:483,12:NetstringReceiverTestCase.test_buffer: Invalid name "packet_size" (should match ([a-z_][a-zA-Z0-9]*)$)
       C0103:486,12:NetstringReceiverTestCase.test_buffer: Invalid name "MAX_LENGTH" (should match ((([a-z_])|([a-z]+_[a-z]))[a-zA-Z0-9]+)$)
       C0301:705,0: Line too long (80/79)
       W9202:809,4:RecvdAttributeMixin.makeMessage: Missing epytext markup @param for argument "data"
       W9202:809,4:RecvdAttributeMixin.makeMessage: Missing epytext markup @param for argument "protocol"
       W9204:809,4:RecvdAttributeMixin.makeMessage: Missing epytext markup @return for return value
       C0103:863,8:RecvdAttributeMixin.test_switching: Invalid name "SWITCH" (should match ([a-z_][a-zA-Z0-9]*)$)
       C0103:890,8:RecvdAttributeMixin.test_recvdInLengthLimitExceeded: Invalid name "DATA" (should match ([a-z_][a-zA-Z0-9]*)$)
       ```
    4. I guess introom's empty misc newsfile didn't get applied from the
       patch. Remember to create it before merging.
    5. You mentioned  in a previous comment that recvd is deprecated.
       I found ticket [[#5770](https://github.com/twisted/twisted/issues/5770)](https://github.com/twisted/twisted/issues/5770) is still open but almost ready to be
       merged.  Might be worth adding a Refs: #5770 in the commit message.


I'd be happy for this to be merged if you fix the typo and the line
length and whitespace issues above.

The other twistedchecker errors would be nice to fix (invalid variable
names, missing mixin docstrings) but it may not be a good use of time.



Up to you.

-RichardW.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_23"></a>|@tomprince set status to `closed`|
|-|-|

(In [42224]) Merge coding-standard-6042: Improve coding standard in twisted.protocols.test.test_basic.

Author: esacteksab, introom, tom.prince
Reviewer: tom.prince, thijs, rwall
Fixes: [#6042](https://github.com/twisted/twisted/issues/6042)
